src/agent.h

```diff
@@ -41,6 +41,9 @@ class MetadataAgent {
   // Starts serving.
   void start();
 
+  // Stops serving.
+  void stop();
```

Comment:
Let's renaming this to `notifyStop`, or something similar to ensure that it's clear that this simply signals threads that we're shutting down.